shared/modules/transaction.utils.ts

```diff
@@ -325,10 +327,15 @@ function extractLargeMessageValue(dataToParse: string): string | undefined {
  * @param dataToParse
  * @returns
  */
-export const parseTypedDataMessage = (dataToParse: string) => {
-  const result = JSON.parse(dataToParse);
+export const parseTypedDataMessage = (dataToParse: DataMessageParam) => {
+  if (typeof dataToParse === 'object') {
+    return dataToParse;
```

Comment:
We don't want to return if it's an object, but just skip the `JSON.parse` and continue.